src/main.rs

```diff
@@ -35,20 +41,63 @@ fn main() -> Result<(), Error> {
     );
 
     let opt = Opt::from_args();
-    let config: Config = toml::from_str(&fs::read_to_string(opt.config)?)?;
+    let mut config: Config = toml::from_str(&fs::read_to_string(opt.config)?)?;
+
+    match config.validate() {
+        Ok(_) => (),
+        Err(e) => return Err(Error::ValidationError(e)),
+    };
+    for camera in &config.cameras {
+        match camera.validate() {
+            Ok(_) => (),
+            Err(e) => return Err(Error::ValidationError(e)),
+        };
+    }
```

Comment:
Hm - can validate() call the sub-validators automatically?

Reply:
I did try first but it didn't validate it. Perhaps because it is inside a vector and it is not smart enough to test if the contents are validate-able. Maybe there's an option in the docs though.. I shall read into it more.

Reply:
Honestly don't worry about it for this PR. We can clean up if desired. Let's land the functionality.

Reply:
I have it sorted out anyways here just add `#[validate]`